isotope/convert/pkg/graph/validation.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this currentFile except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package graph
+
+import (
+	"errors"
+	"fmt"
+
+	"istio.io/tools/isotope/convert/pkg/graph/script"
+)
+
+// validate returns nil if g is valid.
+// g is valid if a ServiceGraph:
+// - Each of its services only makes requests to other defined services.
+// - ConcurrentCommands do not contain other ConcurrentCommands.
+func validate(g ServiceGraph) (err error) {
+	svcNames := map[string]bool{}
+	for _, svc := range g.Services {
+		svcNames[svc.Name] = true
+	}
+	for _, svc := range g.Services {
+		err = validateCommands(svc.Script, svcNames)
+		if err != nil {
+			return
+		}
+	}
+	return
```

Comment:
nit: avoid named returns 
```
if err:= validateCommands(svc.Script, svcNames); err != nil {
    return err
}
```

nit:  prefer `return nil`  so it clear that there is no error

Reply:
Done.